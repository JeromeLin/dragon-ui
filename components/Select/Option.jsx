import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Menu from '../Menu';

class Option extends Component {
  render() {
    const { props } = this;
    const {
      children, checked, disabled, isDisabled
    } = props;

    const disableProps = {
      isDisabled,
    };
    if (disabled) {
      disableProps.disabled = true;
    }
    return (
      <Menu.Item
        checked={checked}
        {...disableProps}
        onClick={e => props.onChange(e)}
      >
        {children}
      </Menu.Item>
    );
  }
}

Option.propTypes = {
  defaultChecked: PropTypes.bool,
  isDisabled: PropTypes.bool,
  onChange: PropTypes.func
};

Option.defaultProps = {
  defaultChecked: false,
  isDisabled: false,
  onChange: () => {}
};

export default Option;
