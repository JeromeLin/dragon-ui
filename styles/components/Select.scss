@mixin select-size($height, $padding, $font-size, $border-radius) {  
  .ui-select-selection {
    height: $height;
    font-size: $font-size;
    border-radius: $border-radius;

    .ui-select-text {
      line-height: $height - 2;
      padding-left: $padding;
      padding-right: 28px;
    }

    .ui-select-icon,
    .ui-select-arrow {
      right: $padding;
      margin-top:  - $font-size / 2;
    }
  }

  .ui-dropdown {
    position: absolute;
    left: 0;
    top: $height + 4;
  }
}

.ui-select {
  @include select-size($base-height, $base-padding, $base-fontSize, $base-radius);
  display: inline-block;
  margin: 0;
  position: relative;
  vertical-align: middle;
  color: #666;

  .ui-select-selection {
    outline: none;
    user-select: none;
    display: block;
    background-color: #fff;
    border-radius: 6px;
    border: 1px solid #d9d9d9;
    transition: all .3s cubic-bezier(.645, .045, .355, 1);
    /*box-shadow: inset 0 1px 3px #e3e3e3;*/
    cursor: pointer;

    &:hover {
      border-color: #57c5f7;
    }

    &:active {
      border-color: #57c5f7;
      outline: 0;
      box-shadow: inset 0 1px 1px rgba(0, 0, 0, .075), 0 0 8px rgba(102, 175, 233, .6);
    }

    .ui-select-text {
      display: block;
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;

      &.ui-select-text-placeholder {
        color: #ccc;
      }
    }

    .ui-select-icon,
    .ui-select-arrow {
      position: absolute;
      top: 50%;
      color: #d9d9d9;
      transition: transform .2s ease;
      line-height: normal;
    }
  }

  &.ui-select-open {
    .ui-select-selection {
      border-color: #57c5f7;
      outline: 0;
      box-shadow: inset 0 1px 1px rgba(0, 0, 0, .075), 0 0 8px rgba(102, 175, 233, .6);

      .ui-select-arrow {
        transform: rotate(180deg);
      }
    }
  }

  &.ui-select-disabled {

    .ui-select-selection {
      cursor: not-allowed;
      background-color: #f3f5f7;

      &:hover {
        border-color: #d9d9d9;
      }

      &:active {
        border-color: #d9d9d9;
      }
    }
  }

  &.size-xl {
    @include select-size($base-height-xl, $base-padding-xl, $base-fontSize-xl, $base-radius-xl);
  }
  &.size-lg {
    @include select-size($base-height-lg, $base-padding-lg, $base-fontSize-lg, $base-radius-lg);
  }
  &.size-sm {
    @include select-size($base-height-sm, $base-padding-sm, $base-fontSize-sm, $base-radius-sm);
  }
  &.size-xs {
    @include select-size($base-height-xs, $base-padding-xs, $base-fontSize-xs, $base-radius-xs);
  }

}