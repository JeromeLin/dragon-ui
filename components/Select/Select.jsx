
import React, { Component, PropTypes } from 'react';
import ReactDOM from 'react-dom';
import classnames from 'classnames';
import Events from '../utils/events';
import isNodeInTree from '../utils/isNodeInTree';
import Option from './Option';
import Dropdown from '../Dropdown';
import Input from '../Input';
import Menu from '../Menu';
import Icon from '../Icon';

class Select extends Component {

  constructor(props) {
    super(props);
    this.unmounted = false;
    this.state = {
      value      : props.value || props.defaultValue || this.getCheckedValue(props.children),
      dropdown   : false,
      searchValue: '',
    };
  }

  componentDidMount() {
    this.unmounted = true;
  }

  componentWillReceiveProps(nextProps) {
    if ('value' in nextProps || this.getCheckedValue(nextProps.children)) {
      this.setState({
        value: nextProps.value || this.getCheckedValue(nextProps.children)
      });
    }
  }

  componentWillUnmount() {
    this.unmounted = false;
    this.unbindOuterHandlers();
  }

  render () {
    const props = this.props;
    const { placeholder, isRadius, isDisabled, isSearch, size, onChange, ...others } = props;
    const disabled = 'disabled' in props || isDisabled;
    const radius = 'radius' in props || isRadius;
    const search = 'search' in props || isSearch;

    let valueText = placeholder,
        hasValue = false;

    let children = React.Children.map(props.children, (option, index) => {
      if (this.state.value == option.props.value) {
        valueText = option.props.children;
        hasValue = true;
      }

      if (search && option.props.children.toString().indexOf(this.state.searchValue) < 0) {
        return null;
      }

      return (
        <Option
          {...option.props}
          onChange={(e) => this.onOptionChange(e, option.props, index)}
          checked={this.state.value === option.props.value} />
      );
    });

    let menus = (children.length > 0)
              ? <Menu size={size}>{children}</Menu>
              : <span className="ui-select-notfound">没有找到数据</span>;

    const searchInput = search
                      ? (
                          <div className="ui-select-input">
                            <Input isRadius={radius} size="sm" placeholder="输入关键字" value={this.state.searchValue} onChange={(e) => {
                              let searchValue = e.target.value;
                              this.setState({searchValue})
                            }} />
                          </div>
                        )
                      : null;

    const cls = classnames({
      'ui-select'     : true,
      'ui-select-open': this.state.dropdown,
      'disabled'      : disabled,
      'radius'        : radius,
      [`size-${size}`]: !!size,
    });

    const textCls = classnames({
      'ui-select-text'            : true,
      'ui-select-text-placeholder': !hasValue,
    });

    return (
      <span className={cls} {...others}>
        <span className="ui-select-selection" role="combobox" aria-autocomplete="list" aria-haspopup="true" aria-expanded="false" onClick={(e) => !disabled && this.onSelectClick(e)}>
          <span className={textCls}>{valueText}</span>
          <Icon type="arrow-bottom" className="ui-select-arrow" />
        </span>
        <Dropdown visible={this.state.dropdown} isRadius={radius}>
          {searchInput}
          {menus}
        </Dropdown>
      </span>
    );
  }

  getCheckedValue(children) {
    let checkedValue = null;
    React.Children.forEach(children, (option) => {
      if (option.props && option.props.checked) {
        checkedValue = option.props.value;
      }
    });
    return checkedValue;
  }

  onSelectClick(e) {
    e.preventDefault();
    this.setDropdown(!this.state.dropdown);
  }

  onOptionChange(e, props, index) {
    if ('disabled' in props || props.isDisabled) {
      return;
    }

    this.setState({
      value      : props.value,
      searchValue: '',
    });

    const selected = {
      index: index,
      value: props.value,
      text : props.children,
    };
    this.setDropdown(false, this.props.onChange(selected));
  }

  setDropdown(isOpen, callback) {
    if (!this.unmounted) return;

    if (isOpen) {
      this.bindOuterHandlers();
    } else {
      this.unbindOuterHandlers();
    }

    this.setState({
      dropdown: isOpen
    }, () => {
      callback && callback();
    });
  }

  handleKeyup(e) {
    (e.keyCode === 27) && this.setDropdown(false);
  }

  handleOuterClick(e) {
    if (!this.unmounted || isNodeInTree(e.target, ReactDOM.findDOMNode(this))) {
      return;
    }
    this.setDropdown(false);
  }

  bindOuterHandlers() {
    Events.on(document, 'click', (e) => this.handleOuterClick(e));
    Events.on(document, 'keyup', (e) => this.handleKeyup(e));
  }

  unbindOuterHandlers() {
    Events.off(document, 'click', (e) => this.handleOuterClick(e));
    Events.off(document, 'keyup', (e) => this.handleKeyup(e));
  }
}

Select.propTypes = {
  isRadius      : PropTypes.bool,
  isDisabled    : PropTypes.bool,
  isSearch      : PropTypes.bool,
  onChange      : PropTypes.func,
};

Select.defaultProps = {
  isRadius      : false,
  isDisabled    : false,
  isSearch      : false,
  onChange      : () => {},
};

export default Select;