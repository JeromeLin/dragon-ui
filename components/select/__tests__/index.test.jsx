import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Select from '../index';

const { Option } = Select;

describe('Select', () => {
  it('Select', () => {
    const wrapper = render(
      <Select>
        <Option value="a">我是A</Option>
        <Option value="b">我是B</Option>
      </Select>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render defaultValue correctly ', () => {
    const wrapper = mount(
      <Select defaultValue="lily">
        <Option value="jack">Jack</Option>
        <Option value="lucy">Lucy</Option>
        <Option value="lily">Lily</Option>
      </Select>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Select disabled', () => {
    const wrapper = mount(
      <Select disabled>
        <Option value="a">我是A</Option>
        <Option value="b">我是B</Option>
      </Select>,
    );

    wrapper.find('.zw-tag-input-box').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('multiple Select init value', () => {
    const wrapper = mount(
      <Select multiple defaultValue={['jack', 'lucy']}>
        <Option value="jack">Jack</Option>
        <Option value="lucy">Lucy</Option>
        <Option value="lily">Lily</Option>
        <Option value="john">John</Option>
        <Option value="honey">Honey</Option>
      </Select>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
