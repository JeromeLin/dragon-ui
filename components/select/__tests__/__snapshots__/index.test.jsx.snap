// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Select 1`] = `
<div
  style="display:inline-block"
>
  <span>
    <div
      class="zw-tag-input-box is-radius"
    >
      <div
        class="value-text"
        style="display:inline-block"
      />
      <span
        class="zw-tag-input__div-placeholder"
        style="display:inline-block"
      >
        请选择
      </span>
      <i
        class="zw-icon arrow-bottom zw-icon--arrow-bottom zw-icon--default zw-icon--md"
        style="font-size:initial"
        theme="default"
        type="arrow-bottom"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 32 32"
          width="1em"
        >
          <use
            href="#arrow-bottom"
          />
        </svg>
      </i>
    </div>
  </span>
</div>
`;

exports[`Select Select disabled 1`] = `
<ForwardRef(forwardRef)
  disabled={true}
>
  <LocaleReceiver
    disabled={true}
    forwardedRef={null}
  >
    <Select
      disabled={true}
      locale={
        Object {
          "noMatch": "没有找到数据",
          "placeholder": "请选择",
          "searchPlaceholder": "输入关键字",
        }
      }
      localeCode="zh-cn"
      onChange={[Function]}
      onSearchChange={[Function]}
      prefixCls="zw-select"
      radius={true}
    >
      <Dropdown
        content={
          <Menu
            defaultOpenKeys={Array []}
            defaultSelectedKeys={Array []}
            inlineCollapsed={false}
            inlineIndent={12}
            mode="inline"
            onOpenChange={[Function]}
            onSelect={[Function]}
            prefixCls="zw-menu"
            style={
              Object {
                "maxHeight": 250,
                "overflow": "auto",
              }
            }
            theme="light"
          >
            <Option
              checked={false}
              isDisabled={false}
              onChange={[Function]}
              value="a"
            >
              我是A
            </Option>
            <Option
              checked={false}
              isDisabled={false}
              onChange={[Function]}
              value="b"
            >
              我是B
            </Option>
          </Menu>
        }
        direction="bottomLeft"
        disabled={true}
        onVisibleChange={[Function]}
        prefixCls="zw-dropdown"
        shape="radius"
        trigger="click"
        visible={false}
      >
        <Popper
          animationDuration={300}
          animationType="zoomFade"
          arrowPointAtCenter={false}
          className="zw-dropdown zw-dropdown--radius"
          content={
            <div
              onKeyDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              tabIndex={-1}
            >
              <Menu
                defaultOpenKeys={Array []}
                defaultSelectedKeys={Array []}
                inlineCollapsed={false}
                inlineIndent={12}
                mode="inline"
                onOpenChange={[Function]}
                onSelect={[Function]}
                prefixCls="zw-menu"
                style={
                  Object {
                    "maxHeight": 250,
                    "overflow": "auto",
                  }
                }
                theme="light"
              >
                <Option
                  checked={false}
                  isDisabled={false}
                  onChange={[Function]}
                  value="a"
                >
                  我是A
                </Option>
                <Option
                  checked={false}
                  isDisabled={false}
                  onChange={[Function]}
                  value="b"
                >
                  我是B
                </Option>
              </Menu>
            </div>
          }
          destroy={true}
          direction="bottomLeft"
          hasArrow={false}
          mouseEnterDelay={150}
          mouseLeaveDelay={100}
          onVisibleChange={[Function]}
          prefixCls="za-popper"
          trigger="manual"
          visible={false}
        >
          <div
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <span
              onClick={[Function]}
              onContextMenu={[Function]}
              onKeyDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <InputWithTags
                active={false}
                disabled={true}
                onDeleteTag={[Function]}
                onSearchChange={[Function]}
                placeholder="请选择"
                radius={true}
                search={false}
                searchValue=""
              >
                <div
                  className="zw-tag-input-box is-radius is-disabled"
                  onClick={[Function]}
                >
                  <div
                    className="value-text"
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  />
                  <span
                    className="zw-tag-input__div-placeholder"
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    请选择
                  </span>
                  <Icon
                    className="arrow-bottom"
                    prefixCls="zw-icon"
                    size="md"
                    style={
                      Object {
                        "fontSize": "initial",
                      }
                    }
                    theme="default"
                    type="arrow-bottom"
                    viewBox="0 0 32 32"
                  >
                    <Iconfont
                      className="zw-icon arrow-bottom zw-icon--arrow-bottom zw-icon--default zw-icon--md"
                      size="md"
                      style={
                        Object {
                          "fontSize": "initial",
                        }
                      }
                      theme="default"
                      type="arrow-bottom"
                    >
                      <Icon
                        className="zw-icon arrow-bottom zw-icon--arrow-bottom zw-icon--default zw-icon--md"
                        prefixCls="zw-icon"
                        size="md"
                        style={
                          Object {
                            "fontSize": "initial",
                          }
                        }
                        theme="default"
                        type="arrow-bottom"
                        viewBox="0 0 32 32"
                      >
                        <i
                          className="zw-icon arrow-bottom zw-icon--arrow-bottom zw-icon--default zw-icon--md"
                          size="md"
                          style={
                            Object {
                              "fontSize": "initial",
                            }
                          }
                          theme="default"
                          type="arrow-bottom"
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            viewBox="0 0 32 32"
                            width="1em"
                          >
                            <use
                              xlinkHref="#arrow-bottom"
                            />
                          </svg>
                        </i>
                      </Icon>
                    </Iconfont>
                  </Icon>
                </div>
              </InputWithTags>
            </span>
          </div>
        </Popper>
      </Dropdown>
    </Select>
  </LocaleReceiver>
</ForwardRef(forwardRef)>
`;

exports[`Select multiple Select init value 1`] = `
<ForwardRef(forwardRef)
  defaultValue={
    Array [
      "jack",
      "lucy",
    ]
  }
  multiple={true}
>
  <LocaleReceiver
    defaultValue={
      Array [
        "jack",
        "lucy",
      ]
    }
    forwardedRef={null}
    multiple={true}
  >
    <Select
      defaultValue={
        Array [
          "jack",
          "lucy",
        ]
      }
      locale={
        Object {
          "noMatch": "没有找到数据",
          "placeholder": "请选择",
          "searchPlaceholder": "输入关键字",
        }
      }
      localeCode="zh-cn"
      multiple={true}
      onChange={[Function]}
      onSearchChange={[Function]}
      prefixCls="zw-select"
      radius={true}
    >
      <Dropdown
        content={
          <Menu
            defaultOpenKeys={Array []}
            defaultSelectedKeys={Array []}
            inlineCollapsed={false}
            inlineIndent={12}
            mode="inline"
            onOpenChange={[Function]}
            onSelect={[Function]}
            prefixCls="zw-menu"
            style={
              Object {
                "maxHeight": 250,
                "overflow": "auto",
              }
            }
            theme="light"
          >
            <Option
              checked={true}
              isDisabled={false}
              onChange={[Function]}
              showCheckIcon={true}
              value="jack"
            >
              Jack
            </Option>
            <Option
              checked={true}
              isDisabled={false}
              onChange={[Function]}
              showCheckIcon={true}
              value="lucy"
            >
              Lucy
            </Option>
            <Option
              checked={false}
              isDisabled={false}
              onChange={[Function]}
              showCheckIcon={true}
              value="lily"
            >
              Lily
            </Option>
            <Option
              checked={false}
              isDisabled={false}
              onChange={[Function]}
              showCheckIcon={true}
              value="john"
            >
              John
            </Option>
            <Option
              checked={false}
              isDisabled={false}
              onChange={[Function]}
              showCheckIcon={true}
              value="honey"
            >
              Honey
            </Option>
          </Menu>
        }
        direction="bottomLeft"
        disabled={false}
        onVisibleChange={[Function]}
        prefixCls="zw-dropdown"
        shape="radius"
        trigger="click"
        visible={false}
      >
        <Popper
          animationDuration={300}
          animationType="zoomFade"
          arrowPointAtCenter={false}
          className="zw-dropdown zw-dropdown--radius"
          content={
            <div
              onKeyDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              tabIndex={-1}
            >
              <Menu
                defaultOpenKeys={Array []}
                defaultSelectedKeys={Array []}
                inlineCollapsed={false}
                inlineIndent={12}
                mode="inline"
                onOpenChange={[Function]}
                onSelect={[Function]}
                prefixCls="zw-menu"
                style={
                  Object {
                    "maxHeight": 250,
                    "overflow": "auto",
                  }
                }
                theme="light"
              >
                <Option
                  checked={true}
                  isDisabled={false}
                  onChange={[Function]}
                  showCheckIcon={true}
                  value="jack"
                >
                  Jack
                </Option>
                <Option
                  checked={true}
                  isDisabled={false}
                  onChange={[Function]}
                  showCheckIcon={true}
                  value="lucy"
                >
                  Lucy
                </Option>
                <Option
                  checked={false}
                  isDisabled={false}
                  onChange={[Function]}
                  showCheckIcon={true}
                  value="lily"
                >
                  Lily
                </Option>
                <Option
                  checked={false}
                  isDisabled={false}
                  onChange={[Function]}
                  showCheckIcon={true}
                  value="john"
                >
                  John
                </Option>
                <Option
                  checked={false}
                  isDisabled={false}
                  onChange={[Function]}
                  showCheckIcon={true}
                  value="honey"
                >
                  Honey
                </Option>
              </Menu>
            </div>
          }
          destroy={true}
          direction="bottomLeft"
          hasArrow={false}
          mouseEnterDelay={150}
          mouseLeaveDelay={100}
          onVisibleChange={[Function]}
          prefixCls="za-popper"
          trigger="manual"
          visible={false}
        >
          <div
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <span
              onClick={[Function]}
              onContextMenu={[Function]}
              onKeyDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <InputWithTags
                active={false}
                disabled={false}
                onDeleteTag={[Function]}
                onSearchChange={[Function]}
                placeholder="请选择"
                radius={true}
                search={false}
                searchValue=""
                value={
                  Array [
                    Object {
                      "key": "jack",
                      "value": "Jack",
                    },
                    Object {
                      "key": "lucy",
                      "value": "Lucy",
                    },
                  ]
                }
              >
                <div
                  className="zw-tag-input-box is-radius"
                  onClick={[Function]}
                >
                  <div
                    className="zw-tag-list-box"
                    key="jack"
                  >
                    <Tag
                      bordered={true}
                      closable={true}
                      key="jack"
                      onClick={[Function]}
                      onClose={[Function]}
                      prefixCls="zw-tag"
                      style={
                        Object {
                          "height": 22,
                          "lineHeight": "22px",
                          "maxWidth": 100,
                        }
                      }
                    >
                      <div
                        className="zw-tag"
                        onClick={[Function]}
                        style={
                          Object {
                            "height": 22,
                            "lineHeight": "22px",
                            "maxWidth": 100,
                          }
                        }
                      >
                        Jack
                        <Icon
                          onClick={[Function]}
                          prefixCls="zw-icon"
                          size="md"
                          theme="default"
                          type="wrong"
                          viewBox="0 0 32 32"
                        >
                          <Iconfont
                            className="zw-icon zw-icon--wrong zw-icon--default zw-icon--md"
                            onClick={[Function]}
                            size="md"
                            theme="default"
                            type="wrong"
                          >
                            <Icon
                              className="zw-icon zw-icon--wrong zw-icon--default zw-icon--md"
                              onClick={[Function]}
                              prefixCls="zw-icon"
                              size="md"
                              theme="default"
                              type="wrong"
                              viewBox="0 0 32 32"
                            >
                              <i
                                className="zw-icon zw-icon--wrong zw-icon--default zw-icon--md"
                                onClick={[Function]}
                                size="md"
                                theme="default"
                                type="wrong"
                              >
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  viewBox="0 0 32 32"
                                  width="1em"
                                >
                                  <use
                                    xlinkHref="#wrong"
                                  />
                                </svg>
                              </i>
                            </Icon>
                          </Iconfont>
                        </Icon>
                      </div>
                    </Tag>
                  </div>
                  <div
                    className="zw-tag-list-box"
                    key="lucy"
                  >
                    <Tag
                      bordered={true}
                      closable={true}
                      key="lucy"
                      onClick={[Function]}
                      onClose={[Function]}
                      prefixCls="zw-tag"
                      style={
                        Object {
                          "height": 22,
                          "lineHeight": "22px",
                          "maxWidth": 100,
                        }
                      }
                    >
                      <div
                        className="zw-tag"
                        onClick={[Function]}
                        style={
                          Object {
                            "height": 22,
                            "lineHeight": "22px",
                            "maxWidth": 100,
                          }
                        }
                      >
                        Lucy
                        <Icon
                          onClick={[Function]}
                          prefixCls="zw-icon"
                          size="md"
                          theme="default"
                          type="wrong"
                          viewBox="0 0 32 32"
                        >
                          <Iconfont
                            className="zw-icon zw-icon--wrong zw-icon--default zw-icon--md"
                            onClick={[Function]}
                            size="md"
                            theme="default"
                            type="wrong"
                          >
                            <Icon
                              className="zw-icon zw-icon--wrong zw-icon--default zw-icon--md"
                              onClick={[Function]}
                              prefixCls="zw-icon"
                              size="md"
                              theme="default"
                              type="wrong"
                              viewBox="0 0 32 32"
                            >
                              <i
                                className="zw-icon zw-icon--wrong zw-icon--default zw-icon--md"
                                onClick={[Function]}
                                size="md"
                                theme="default"
                                type="wrong"
                              >
                                <svg
                                  fill="currentColor"
                                  height="1em"
                                  viewBox="0 0 32 32"
                                  width="1em"
                                >
                                  <use
                                    xlinkHref="#wrong"
                                  />
                                </svg>
                              </i>
                            </Icon>
                          </Iconfont>
                        </Icon>
                      </div>
                    </Tag>
                  </div>
                  <Icon
                    className="arrow-bottom"
                    prefixCls="zw-icon"
                    size="md"
                    style={
                      Object {
                        "fontSize": "initial",
                      }
                    }
                    theme="default"
                    type="arrow-bottom"
                    viewBox="0 0 32 32"
                  >
                    <Iconfont
                      className="zw-icon arrow-bottom zw-icon--arrow-bottom zw-icon--default zw-icon--md"
                      size="md"
                      style={
                        Object {
                          "fontSize": "initial",
                        }
                      }
                      theme="default"
                      type="arrow-bottom"
                    >
                      <Icon
                        className="zw-icon arrow-bottom zw-icon--arrow-bottom zw-icon--default zw-icon--md"
                        prefixCls="zw-icon"
                        size="md"
                        style={
                          Object {
                            "fontSize": "initial",
                          }
                        }
                        theme="default"
                        type="arrow-bottom"
                        viewBox="0 0 32 32"
                      >
                        <i
                          className="zw-icon arrow-bottom zw-icon--arrow-bottom zw-icon--default zw-icon--md"
                          size="md"
                          style={
                            Object {
                              "fontSize": "initial",
                            }
                          }
                          theme="default"
                          type="arrow-bottom"
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            viewBox="0 0 32 32"
                            width="1em"
                          >
                            <use
                              xlinkHref="#arrow-bottom"
                            />
                          </svg>
                        </i>
                      </Icon>
                    </Iconfont>
                  </Icon>
                </div>
              </InputWithTags>
            </span>
          </div>
        </Popper>
      </Dropdown>
    </Select>
  </LocaleReceiver>
</ForwardRef(forwardRef)>
`;

exports[`Select render defaultValue correctly  1`] = `
<ForwardRef(forwardRef)
  defaultValue="lily"
>
  <LocaleReceiver
    defaultValue="lily"
    forwardedRef={null}
  >
    <Select
      defaultValue="lily"
      locale={
        Object {
          "noMatch": "没有找到数据",
          "placeholder": "请选择",
          "searchPlaceholder": "输入关键字",
        }
      }
      localeCode="zh-cn"
      onChange={[Function]}
      onSearchChange={[Function]}
      prefixCls="zw-select"
      radius={true}
    >
      <Dropdown
        content={
          <Menu
            defaultOpenKeys={Array []}
            defaultSelectedKeys={Array []}
            inlineCollapsed={false}
            inlineIndent={12}
            mode="inline"
            onOpenChange={[Function]}
            onSelect={[Function]}
            prefixCls="zw-menu"
            style={
              Object {
                "maxHeight": 250,
                "overflow": "auto",
              }
            }
            theme="light"
          >
            <Option
              checked={false}
              isDisabled={false}
              onChange={[Function]}
              value="jack"
            >
              Jack
            </Option>
            <Option
              checked={false}
              isDisabled={false}
              onChange={[Function]}
              value="lucy"
            >
              Lucy
            </Option>
            <Option
              checked={true}
              isDisabled={false}
              onChange={[Function]}
              value="lily"
            >
              Lily
            </Option>
          </Menu>
        }
        direction="bottomLeft"
        disabled={false}
        onVisibleChange={[Function]}
        prefixCls="zw-dropdown"
        shape="radius"
        trigger="click"
        visible={false}
      >
        <Popper
          animationDuration={300}
          animationType="zoomFade"
          arrowPointAtCenter={false}
          className="zw-dropdown zw-dropdown--radius"
          content={
            <div
              onKeyDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              tabIndex={-1}
            >
              <Menu
                defaultOpenKeys={Array []}
                defaultSelectedKeys={Array []}
                inlineCollapsed={false}
                inlineIndent={12}
                mode="inline"
                onOpenChange={[Function]}
                onSelect={[Function]}
                prefixCls="zw-menu"
                style={
                  Object {
                    "maxHeight": 250,
                    "overflow": "auto",
                  }
                }
                theme="light"
              >
                <Option
                  checked={false}
                  isDisabled={false}
                  onChange={[Function]}
                  value="jack"
                >
                  Jack
                </Option>
                <Option
                  checked={false}
                  isDisabled={false}
                  onChange={[Function]}
                  value="lucy"
                >
                  Lucy
                </Option>
                <Option
                  checked={true}
                  isDisabled={false}
                  onChange={[Function]}
                  value="lily"
                >
                  Lily
                </Option>
              </Menu>
            </div>
          }
          destroy={true}
          direction="bottomLeft"
          hasArrow={false}
          mouseEnterDelay={150}
          mouseLeaveDelay={100}
          onVisibleChange={[Function]}
          prefixCls="za-popper"
          trigger="manual"
          visible={false}
        >
          <div
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <span
              onClick={[Function]}
              onContextMenu={[Function]}
              onKeyDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <InputWithTags
                active={false}
                disabled={false}
                onDeleteTag={[Function]}
                onSearchChange={[Function]}
                placeholder="请选择"
                radius={true}
                search={false}
                searchValue=""
                value="Lily"
              >
                <div
                  className="zw-tag-input-box is-radius"
                  onClick={[Function]}
                >
                  <div
                    className="value-text"
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    Lily
                  </div>
                  <Icon
                    className="arrow-bottom"
                    prefixCls="zw-icon"
                    size="md"
                    style={
                      Object {
                        "fontSize": "initial",
                      }
                    }
                    theme="default"
                    type="arrow-bottom"
                    viewBox="0 0 32 32"
                  >
                    <Iconfont
                      className="zw-icon arrow-bottom zw-icon--arrow-bottom zw-icon--default zw-icon--md"
                      size="md"
                      style={
                        Object {
                          "fontSize": "initial",
                        }
                      }
                      theme="default"
                      type="arrow-bottom"
                    >
                      <Icon
                        className="zw-icon arrow-bottom zw-icon--arrow-bottom zw-icon--default zw-icon--md"
                        prefixCls="zw-icon"
                        size="md"
                        style={
                          Object {
                            "fontSize": "initial",
                          }
                        }
                        theme="default"
                        type="arrow-bottom"
                        viewBox="0 0 32 32"
                      >
                        <i
                          className="zw-icon arrow-bottom zw-icon--arrow-bottom zw-icon--default zw-icon--md"
                          size="md"
                          style={
                            Object {
                              "fontSize": "initial",
                            }
                          }
                          theme="default"
                          type="arrow-bottom"
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            viewBox="0 0 32 32"
                            width="1em"
                          >
                            <use
                              xlinkHref="#arrow-bottom"
                            />
                          </svg>
                        </i>
                      </Icon>
                    </Iconfont>
                  </Icon>
                </div>
              </InputWithTags>
            </span>
          </div>
        </Popper>
      </Dropdown>
    </Select>
  </LocaleReceiver>
</ForwardRef(forwardRef)>
`;
